src/Workspaces/Remote/Core/Diagnostics/PerformanceQueue.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote.Diagnostics
+{
+    /// <summary>
+    /// This queue hold onto raw performance data. this type itself is not thread safe. the one who uses this type
+    /// should take care of that.
```

Comment:
:bulb: This thread safety notice would typically be documented like this:

```csharp
/// <summary>
/// This queue holds onto raw performance data.
/// </summary>
/// <threadsafety static="true" instance="false"/>
```

Here's an example:

https://github.com/tunnelvisionlabs/dotnet-threading/blob/3e99a9d13476a1e8224d81f282f3cedad143c1bc/Rackspace.Threading/DelayedTask.cs#L18-L22